src/views/Signup.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from "react";
+import { commitMutation } from 'react-relay'
+import environment from '../config/relay'
+import { Redirect } from "react-router";
+import graphql from 'babel-plugin-relay/macro'
+
+import {Form, Button, Container} from "react-bootstrap";
+
+const mutation = graphql`
+    mutation SignUpMutation($input: UserCreateInput!){
+        userCreate(input: $input){
+            user {
+                id
+            }
+        }
+    }
+`
+
+export default function SignUp() {
+    const [username, setUsername] = useState("");
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [redirect, setRedirect] = useState(false)
+
+  function validateForm() {
+    return email.length > 0 && password.length > 0;
+  }
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    
+    commitMutation(
+        environment,
+        {
+          mutation,
+          variables : {
+            input: {
+                username: username,
+                email: email,
+                password: password
+            }
+        },
+          onCompleted: (response, errors) => {
+            console.log('Response received from server.')
+            setRedirect(true)
+          },
+          onError: err => console.log(err),
+        },
+    );
+  }
+
+  
+  return (
+    redirect ? 
+    <Redirect to="/" /> 
```

Comment:
I think it will be cleaner to use the `withRouter` HOC and programmatically route instead of using a component.